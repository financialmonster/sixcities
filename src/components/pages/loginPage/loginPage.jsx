import React, {useState} from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import {Redirect} from 'react-router-dom';

import Header from '../../header/header.jsx';
import {fetchAuth} from '../../../actions/auth.js';
import Spinner from '../../spinner/spinner.jsx';

const LoginPage = ({isLoggedIn, isAuthLoading, fetchAuth}) => {
    const [emailVal, setEmailVal] = useState(``);
    const [passwordVal, setPasswordVal] = useState(``);

    const formSubmitHandler = (evt, email, password) => {
        evt.preventDefault();

        fetchAuth({
            email,
            password
        });
    }

    if(isLoggedIn) {
        return <Redirect to="/" />;
    }

    if(isAuthLoading) {
        return (
            <div className="page page--gray page--login">
                <Header isMain={false} />
                <Spinner />
            </div>
        );
    }

    return (
        <div className="page page--gray page--login">
            <Header isMain={false} />

            <main className="page__main page__main--login">
                <div className="page__login-container container">
                    <section className="login">
                        <h1 className="login__title">Sign in</h1>
                        <form className="login__form form" action="#" method="post"
                            onSubmit={(evt) => formSubmitHandler(evt, emailVal, passwordVal)}
                        >
                            <div className="login__input-wrapper form__input-wrapper">
                                <label className="visually-hidden">E-mail</label>
                                <input className="login__input form__input" type="email" name="email"
                                    placeholder="Email" value={emailVal}
                                    onChange={(evt) => setEmailVal(evt.target.value)} required
                                />
                            </div>
                            <div className="login__input-wrapper form__input-wrapper">
                                <label className="visually-hidden">Password</label>
                                <input className="login__input form__input" type="password" name="password"
                                    placeholder="Password" value={passwordVal}
                                    onChange={(evt) => setPasswordVal(evt.target.value)} required
                                />
                            </div>
                            <button className="login__submit form__submit button" type="submit">Sign in</button>
                        </form>
                    </section>
                    <section className="locations locations--login locations--current">
                        <div className="locations__item">
                            <a className="locations__item-link" href="#">
                                <span>Amsterdam</span>
                            </a>
                        </div>
                    </section>
                </div>
            </main>
        </div>
    );
}

LoginPage.propTypes = {
    fetchAuth: PropTypes.func.isRequired,
    isLoggedIn: PropTypes.bool.isRequired,
    isAuthLoading: PropTypes.bool.isRequired
}

const mapStateToProps = ({auth: {isLoggedIn, isAuthLoading}}) => {
    return {
        isLoggedIn,
        isAuthLoading
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        fetchAuth: ({email, password}) => dispatch(fetchAuth({email, password}))
    } 
}

export {LoginPage};
export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
