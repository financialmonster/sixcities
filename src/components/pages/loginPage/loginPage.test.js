import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {BrowserRouter as Router} from 'react-router-dom';

import {LoginPage} from './loginPage.jsx';

describe(`Correctly renders the LoginPage component `, () => {
    it(`when isLoggedIn property is equal true`, () => {
        const mockStore = configureStore([]);
        const store = mockStore();

        const tree = renderer.create(
            <Provider store={store}>
                <Router>
                    <LoginPage isLoggedIn={true} isAuthLoading={false} isAuthError={false}fetchAuth={() => {}} />
                </Router>
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthLoading property is equal true and both isLoggedIn and isAuthError properties are equal
        false`,
    () => {
        const mockStore = configureStore([]);
        const store = mockStore({auth: {isLoggedIn: false, isAuthStatusLoading: false , isAuthStatusError: false}});

        const tree = renderer.create(
            <Provider store={store}>
                <LoginPage isLoggedIn={false} isAuthLoading={true} isAuthError={false} fetchAuth={() => {}} />
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthError property is equal true and both isLoggedIn and isAuthLoading properties are equal
        false`,
    () => {
        const mockStore = configureStore([]);
        const store = mockStore({auth: {isLoggedIn: false, isAuthStatusLoading: false , isAuthStatusError: false}});

        const tree = renderer.create(
            <Provider store={store}>
                <LoginPage isLoggedIn={false} isAuthLoading={false} isAuthError={true} fetchAuth={() => {}} />
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });

    it(`when isAuthError, isLoggedIn and isAuthLoading properties are equal false`, () => {
        const mockStore = configureStore([]);
        const store = mockStore({auth: {isLoggedIn: false, isAuthStatusLoading: false , isAuthStatusError: false}});

        const tree = renderer.create(
            <Provider store={store}>
                <Router>
                    <LoginPage isLoggedIn={false} isAuthLoading={false} isAuthError={false} fetchAuth={() => {}} />
                </Router>
            </Provider>)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
