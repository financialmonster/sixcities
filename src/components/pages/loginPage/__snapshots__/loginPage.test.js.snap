// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders the LoginPage component  when isAuthError property is equal true and both isLoggedIn and isAuthLoading properties are equal
        false 1`] = `
<div
  className="cities__status-wrapper tabs__content"
>
  <b
    className="cities__status"
  >
    We're sorry
  </b>
  <p
    className="cities__status-description"
  >
    It's been an error during the process of 
    user's authorization
    . Please, reload the page and try again
  </p>
</div>
`;

exports[`Correctly renders the LoginPage component  when isAuthError, isLoggedIn and isAuthLoading properties are equal false 1`] = `
<div
  className="page page--gray page--login"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
            title="To the main page"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/login"
                onClick={[Function]}
                title="To the login page"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          action="#"
          className="login__form form"
          method="post"
          onSubmit={[Function]}
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <button
            className="login__submit form__submit button"
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </section>
    </div>
  </main>
</div>
`;

exports[`Correctly renders the LoginPage component  when isAuthLoading property is equal true and both isLoggedIn and isAuthError properties are equal
        false 1`] = `
<div
  className="spinner"
>
  <div
    className="spinnerContent"
  >
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;

exports[`Correctly renders the LoginPage component  when isLoggedIn property is equal true 1`] = `null`;
